File: iosApp/iosApp/BuildNetwork.swift
import Shared

/// Maps the build's `WalletNetwork` Info.plist value onto the core's network
/// enum (U7, R2). The value comes from the per-configuration `WALLET_NETWORK`
/// build setting in `project.yml` — Debug is Mutinynet, Release is mainnet.
///
/// Pure and separate from `WalletModel` so the configuration → network decision
/// is unit-testable; "a shipped build is on mainnet" is worth asserting rather
/// than eyeballing.
///
/// An unrecognized or missing value resolves to mainnet. A wallet wrongly on
/// mainnet refuses test-network invoices, while one wrongly on a test network
/// could present a worthless address as if it were real — so the safe failure
/// direction is mainnet.
func walletNetworkFor(_ infoPlistValue: String?) -> WalletNetwork {
    switch infoPlistValue?.lowercased() {
    case "mutinynet": return .mutinynet
    default: return .mainnet
    }
}

/// The network this build was compiled for, read from the bundle.
func buildWalletNetwork() -> WalletNetwork {
    walletNetworkFor(
        Bundle.main.object(forInfoDictionaryKey: "WalletNetwork") as? String
    )
}

File: androidApp/src/main/kotlin/zinqq/app/BuildNetwork.kt
package zinqq.app

import uniffi.wallet_core.WalletNetwork

/**
 * Maps the build's `BuildConfig.WALLET_NETWORK` string onto the core's network
 * enum (U6, R2).
 *
 * Kept pure and separate from [WalletHolder] so the build-type → network
 * decision is unit-testable — this module has no instrumentation-test
 * infrastructure, and "Release cannot end up on a test network" is exactly the
 * kind of guarantee that should be asserted rather than eyeballed.
 *
 * An unrecognized value resolves to mainnet. The Gradle side already falls back
 * to the build type's default, so reaching this branch means something is wrong
 * with the build config — and mainnet is the safe answer, since a wallet that
 * wrongly runs mainnet refuses test-network invoices, while one that wrongly
 * runs a test network could present a signet address as if it were real.
 */
fun walletNetworkFor(buildConfigValue: String): WalletNetwork =
    when (buildConfigValue.lowercase()) {
        "mutinynet" -> WalletNetwork.MUTINYNET
        else -> WalletNetwork.MAINNET
    }
